Adicionando navegação responsiva

BrianTKK opened the pull request.

BrianTKK committed:
Add Bootstrap link

BrianTKK committed:
Removendo decoração do link adicionada pelo bootstrap

BrianTKK committed:
Nav bar original comentada

BrianTKK committed:
Logo e Heading

Adicionando logo tamanho 45xAUTOpx
Heading alterado de 2rem para 1.75rem (achei que ficou mais harmonico, vale reverter se necessário)

BrianTKK committed:
Links cabeçalho

Adicionando os links corretos com a classe .menu-link e comentando elemento dropdown(pra uso futuro talvez(

BrianTKK committed:
Centralizando navegação

BrianTKK committed:
Removendo atributos desnecessários da .menu

BrianTKK committed:
Adicionando espaçamento entre links

BrianTKK committed:
Corrigindo fonte alterada pelo bootstrap

BrianTKK committed:
Re-add botão de login

"Criar conta" alterado para "acessar conta" pra melhor coerência caso ja tenha conta criada

BrianTKK committed:
Colocando acessar conta fora da UL para posicionamento correto

BrianTKK committed:
Add padding e voltando logo pra 2rem

BrianTKK committed:
Revert "Colocando acessar conta fora da UL para posicionamento correto"

This reverts commit 11095236ff2e4969e7b5584d0171593286d9bb15.

BrianTKK committed:
form de busca comentado

BrianTKK committed:
Acessar Conta

Botão com margem para centralização do menu

BrianTKK committed:
Update style.css

Corrigindo espaçamento entre a nav e Hero para ficar como no original

BrianTKK committed:
add peso na fonte do botão

BrianTKK committed:
removendo classe nav não utilizada

BrianTKK committed:
Corrigindo cores cabeçalho e posicionamento responsivo

BrianTKK committed:
Alteração em nome de classe conflitando com bootstrap

BrianTKK committed:
Comentando nav antiga do login

BrianTKK committed:
Add bootstrap em login.html

BrianTKK committed:
Add Nav na login.html e deixando o botão ativado

BrianTKK committed:
Add bootstrap às demais paginas

BrianTKK committed:
Add Nav nas demais paginas

amimarinho commented on index.html:
```diff
@@ -9,6 +9,7 @@
       href="https://fonts.googleapis.com/css2?family=Gidole&family=Nunito+Sans:ital,opsz,wght@0,6..12,200..1000;1,6..12,200..1000&display=swap"
       rel="stylesheet"
     />
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.6/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-4Q6Gf2aSP4eDXB8Miphtr37CMZZQ5oXLH2yaXMJ2w8e2ZtHTl7GptT4jmndRuHDT" crossorigin="anonymous">
```
Ao adicionar o bootstrap, ele injeta vários arquivos de estilo no código, entre eles o `reboot.css` e o `rts.css`. Esse reboot adiciona margin e padding em vários elementos e o RTS reconfigura todas as fontes. Exemplo:

| Antes | Depois |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/01b59a68-6a38-4d65-83c3-0b4f445c664f) |  ![image](https://github.com/user-attachments/assets/4039167d-7dd6-4a96-a008-97eb56d13727) |


O que dá pra fazer é ir inspecionando cada elemento CSS e ir desfazendo o CSS que vem do bootstrap que a gente não quer, ex:

|  |  |
| --- |  --- |
| ![image](https://github.com/user-attachments/assets/5c27a4bb-6cf7-42b3-bfaa-d8676183c304) | Exemplo dos arquivos `rfs` e `reboot` afetando o `<h3>` do card.

O que eu faria pra arrumar:

```css
/* Remove o font-size errado do h3 */
.card-list-item > h3 {
  font: 800 1.2rem var(--font-gidole);
}

/* Remove as margens incorretas das fontes e listas*/
ul, ol {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

:is(p, h1, h2, h3) {
  margin: 0;
  padding: 0;
}

```

amimarinho commented on index.html:
```diff
@@ -21,7 +22,7 @@
 
   <body>
-    <nav role="navigation" class="nav">
+    <!-- <nav role="navigation" class="nav">
```
Pode excluir o navbar antigo, só não esquece de apagar tbm as classes CSS dele

amimarinho commented on index.html:
```diff
@@ -81,7 +130,7 @@ <h1 class="heading">Iara Games</h1>
         </a>
       </header>
-      <section class="releases bg-dark" id="lancamentos">
+      <section class="releases bg-dark1" id="lancamentos">
```
Porque a alteração do bg-dark pra bg-dark1, colidia com o do bootstrap?

amimarinho commented on index.html:
```diff
@@ -284,5 +333,6 @@ <h3>Ekorella</h3>
       </div>
       <img src="./assets/img/bushes.png" alt="" class="footer-img" />
     </footer>
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.6/dist/js/bootstrap.bundle.min.js" integrity="sha384-j1CDi7MgGQ12Z7Qab0qlWQ/Qqz24Gc6BM0thvEMVjHnfYGF0rmFCozFSxQBxwHKO" crossorigin="anonymous"></script>
```
Você pode adicionar isso no `<head>` junto com os links adicionando o atributo `defer`. Assim:

```html
<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.6/dist/js/bootstrap.bundle.min.js" integrity="sha384-j1CDi7MgGQ12Z7Qab0qlWQ/Qqz24Gc6BM0thvEMVjHnfYGF0rmFCozFSxQBxwHKO" crossorigin="anonymous" defer></script>
```
Adicionar script no body era uma prática que era utilizada antes da existência do defer e já não é mais recomendada

amimarinho commented on style.css:
```diff
@@ -91,24 +88,28 @@ body {
   align-items: center;
 }
 
+.nav-link{
```
Tá sem espaço entre o colchete e a classe, pra você não se preocupar tanto, tu pode habilitar o format-on-save


1. `ctrl+,`
2. digita "format on save" na busca

![image](https://github.com/user-attachments/assets/c92dbe95-af80-4333-b33d-8f266329fc73)

Ai toda vez que tu der `ctrl+s`  ele formata o CSS , HTML e JS que tá aberto.

amimarinho reviewed (changes requested):
Brian, no geral ficou muito bom, só adicionei algumas sugestões de alteração, assim a gente não vai ter problemas com as próximas alterações, como as do carrossel que ainda vai ser add, por ex.

BrianTKK committed:
refactor: Reestrutura inclusão de JS com defer no head

Altera a localização da chamada do script principal para o <head> com o atributo defer, para uma abordagem mais limpa e não-bloqueante de carregamento.

BrianTKK committed:
refactor: Remove classe CSS não utilizada

Deleta a classe '.nav-logo' do arquivo CSS, melhorando a clareza e a manutenibilidade do código-fonte.

BrianTKK committed:
refactor: Remove elemento nav antigo comentado

Deleta o elemento <nav> antigo e comentado do código HTML, melhorando a clareza e removendo código desnecessário.

BrianTKK committed:
refactor: Remove elementos HTML comentados e não utilizados

Deleta o formulário de busca e o link dropdown que estavam comentados no código HTML, melhorando a clareza e reduzindo o código desnecessário.

BrianTKK committed:
fix(styles): Corrige estilos quebrados após adição do Bootstrap

Ajusta a formatação de .card-list-item, ul, ol, p, h1, h2 e h3 para corrigir problemas visuais e de layout causados pela integração dos estilos padrão do Bootstrap.

BrianTKK committed:
fix(styles): Correção dos títulos h2 no tamanho original

BrianTKK committed:
feat: Adiciona modal interativo com carrossel na hero

Implementa uma funcionalidade de modal com carrossel na seção hero do site, permitindo aos usuários visualizar conteúdo adicional ou imagens de forma interativa e organizada.

BrianTKK committed:
feat: Habilita abertura de modal ao clicar na imagem

Implementa a funcionalidade de exibir um modal ao clicar na imagem do jogo-page

BrianTKK committed:
feat(accessibility): melhora acessibilidade nas páginas principais

Melhora a acessibilidade e a responsividade do site, ajustando o carrossel e centralizando botões no menu responsivo via CSS. Padroniza e detalha os atributos alt das imagens, adiciona aria-label, tabindex e aria-hidden em links e ícones das redes sociais, e utiliza role="presentation" e aria-hidden em imagens decorativas. Aplicando as melhorias em index.html, login.html, listing-games.html e jogo-page.html.
